Lombiq.Tests.UI/Extensions/NavigationUITestContextExtensions.cs

```diff
@@ -142,13 +142,19 @@ public static async Task<T> GoToPageAsync<T>(this UITestContext context, string
         context.GoToPageAsync<OrchardCoreRegistrationPage>();
 
     public static Task<OrchardCoreDashboardPage> GoToDashboardAsync(this UITestContext context) =>
-        context.GoToPageAsync<OrchardCoreDashboardPage>();
+        string.IsNullOrEmpty(context.CustomAdminUrl)
+            ? context.GoToPageAsync<OrchardCoreDashboardPage>()
+            : context.GoToPageAsync<OrchardCoreDashboardPage>(context.CustomAdminUrl);
```

Comment:
Instead of doing these if-elses, add a `GoToAdminPageAsync<T>`, then handle the prefixing there. The `[Url("Admin")]` attributes won't be needed anymore if you just supply the prefix at all times.

Reply:
Remove, yes. All affected calls should go through `GoToDashboardAsync` anyway, so nothing should break.

Reply:
And what do you suggest for extending `GoToFeaturesPageAsync()` and similar? They are using extended [Url("Admin/Features")] attributes. Easiest way would be to create `GoToRelativeFeaturesPage() => GoToAdminRelativeUrl("/Features")` but this way if a biggest method is using GoToFeaturesPage() we have to decide there if it has regular prefix or custom.

Reply:
In exactly the same fashion, you can get rid of that `Url` attribute, and all the others for dashboard pages, if they're used via their own methods like `GoToFeaturesPageAsync`. In them, you can supply the URL to `GoToAdminPageAsync<T>`.

An alternative that I just thought about is that you can keep the attributes, but change `BaseUrl` of the current `AtataContext` before navigation, then change back to the original one, but that's probably hackish.

The consumer won't have to decide, that's the point: What I'm explaining here, is that all of this prefix management should be in a single place, and all such page navigating methods should utilize that. The consuming code shouldn't change at all, hence why `BasicOrchardFeaturesTest` should also work without changes. E.g. `GoToFeaturesPageAsync` will continue to be `public static Task<OrchardCoreFeaturesPage> GoToFeaturesPageAsync(this UITestContext context)`, nothing changes in its signature, and no consumer of it needs to change anything.

Reply:
Yes.